### ISSUE TITLE
MPC cannot pre-cool buildings with large RC 

### ISSUE THREAD

**Issue by YangyangFu** [YangyangFu/dynax#6]
The MPC test on RC models shew a few issues:
1. when RC values are large (from identification), the MPC cannot produce sufficient pre-cooling even when the prediction horizon (**PH**) is 1 day.

When RCs were identified as `[10384.31640625, 499089.09375, 1321535.125, 1.5348844528198242, 0.5000327825546265, 1.000040054321289, 20.119935989379883]`, which led to a MSE cost of 0.078, the MPC couldn't pre-cool the buildings when PH=1 day

When RCs were identified as `[6.9789902e+03, 2.1591113e+04, 1.8807944e+05, 3.4490612e+00, 4.9556872e-01, 9.8289281e-02, 4.6257420e+00]`, which lead to a MSE cost of 0.75, the MPC pre-cool the building in a very reasonable way.

2. when  RC values are large, the larger the PH is, the large cost is from the MPC.
This seems very strange.

The reasons might be

- gradient descent optimizers might not converge for large RC, and thus needs to be fine-tuned
- the larger thermal mass the building has, the longer PH should be used, does this hold ?? 

TODO
- [ ] construct the same MPC using optimization modeling framework such as `cvxpy` so that second-order derivative can be used. This can provide a benchmark to see if the problems are in gradient-based optimizers.
- [ ] if the problem is in gradient-descent optimizers, need retune the optimizer
- [ ] if the problem is not in gradient-descent optimizers, then need to think why MPC cannot work well with large RCs.
